src/external.ml

```diff
@@ -114,3 +120,38 @@ let runExternalProgram cmd =
          else "\n\n" ^ Util.process_status_to_string returnValue)))
       (* Stop typechechecker from complaining about non-exhaustive pattern above *)
       | _ -> assert false)
+
+let runExternalProgramArgs cmd args =
+  if Util.osType = `Win32 && not Util.isCygwin then begin
+    debug (fun()-> Util.msg "Executing external program windows-style\n");
+    let c = openProcessArgsIn cmd args in
```

Comment:
Since the only difference in these two functions is the `openProcessArgs*` call in two locations then I propose an implementation in spirit of this:
```diff
diff --git a/src/external.ml b/src/external.ml
index cefb9ea9..bbe03d48 100644
--- a/src/external.ml
+++ b/src/external.ml
@@ -80,10 +80,10 @@ let readChannelsTillEof l =
        >>= (fun res -> return (String.concat "\n" (Safelist.rev res))))
     l
 
-let runExternalProgram cmd =
+let runExternalProgramAux ~winProc ~posixProc =
   if Util.osType = `Win32 && not Util.isCygwin then begin
     debug (fun()-> Util.msg "Executing external program windows-style\n");
-    let c = openProcessIn ("\"" ^ cmd ^ "\"") in
+    let c = winProc () in
     let log = Util.trimWhitespace (readChannelTillEof c) in
     let returnValue = closeProcessIn c in
     let resultLog =
@@ -95,7 +95,7 @@ let runExternalProgram cmd =
          "") in
     Lwt.return (returnValue, resultLog)
   end else
-    let (out, ipt, err) as desc = openProcessFull cmd in
+    let (out, ipt, err) as desc = posixProc () in
     let out = Lwt_unix.intern_in_channel out in
     let err = Lwt_unix.intern_in_channel err in
     readChannelsTillEof [out;err]
@@ -114,3 +114,13 @@ let runExternalProgram cmd =
          else "\n\n" ^ Util.process_status_to_string returnValue)))
       (* Stop typechechecker from complaining about non-exhaustive pattern above *)
       | _ -> assert false)
+
+let runExternalProgram cmd =
+  runExternalProgramAux
+    ~winProc:(fun () -> openProcessIn ("\"" ^ cmd ^ "\""))
+    ~posixProc:(fun () -> openProcessFull cmd)
+
+let runExternalProgramArgs cmd args =
+  runExternalProgramAux
+    ~winProc:(fun () -> openProcessArgsIn cmd args)
+    ~posixProc:(fun () -> openProcessArgsFull cmd args)
```

Reply:
I've applied this patch; much nicer, thanks!